pkg/storage/mysql/migrator.go

```diff
@@ -0,0 +1,138 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/go-sql-driver/mysql"
+	"github.com/pressly/goose/v3"
+
+	"github.com/openfga/openfga/assets"
+	"github.com/openfga/openfga/pkg/storage"
+)
+
+// MySQLMigrationProvider implements MigrationProvider for MySQL.
+type MySQLMigrationProvider struct{}
+
+// NewMySQLMigrationProvider creates a new MySQL migration provider.
+func NewMySQLMigrationProvider() *MySQLMigrationProvider {
+	return &MySQLMigrationProvider{}
+}
+
+// GetSupportedEngine returns the database engine this provider supports.
+func (m *MySQLMigrationProvider) GetSupportedEngine() string {
+	return "mysql"
+}
+
+// RunMigrations executes MySQL database migrations.
+func (m *MySQLMigrationProvider) RunMigrations(ctx context.Context, config storage.MigrationConfig) error {
+	goose.SetLogger(goose.NopLogger())
```

Comment:
It'd be nice to switch away from goose's global registry + default provider

Reply:
Mostly a drive-by, but instead of using global functions, can instead initialize a new provider using `goose.NewProvider` which should avoid global state entirely.

https://pressly.github.io/goose/documentation/provider/